fem/linearform.cpp

```diff
@@ -99,19 +100,65 @@ void LinearForm::AddInteriorFaceIntegrator(LinearFormIntegrator *lfi)
    interior_face_integs.Append(lfi);
 }
 
-void LinearForm::Assemble()
+static bool LinearIntegratorsCanUseDevice(LinearForm *lf)
```

Comment:
Very minor nitpicks:
```suggestion
static bool LinearFormIntegratorsCanUseDevice(LinearForm &lf)
```

Ideally it would be `const LinearForm &lf`, I don't know if all the member functions that are called are `const` though

Reply:
Turned it into `bool SupportsDevice();`